### ISSUE TITLE
Choose Trigger for Copilot Code Suggestions

### ISSUE THREAD

**Issue by jesse-smith** [rstudio/rstudio#14033]
### TLDR
Add a "Show code suggestions [Manually (Tab)]" option to the Copilot settings pane, as in the Code Completion pane.

### Context
I'm using the Copilot preview integration and have been impressed thus far (both with the LLM and with your integration!). Most of the time, it's quite useful to have Copilot offer suggestions automatically. However, there are times when I'm essentially just staring at a piece of code, trying to make sense of it/plan out next steps/etc, and it's fairly distracting to have Copilot add its two cents. For example, I could be sketching out the logic for a large loop; Copilot is useful for filling in the details. However, it's less useful if I'm not yet sure what those details are, and it can be distracting if I'm in the middle of creating the scaffolding and Copilot keeps moving my code around with multi-line suggestions.

Adding an option to only see suggestions when manually triggered would, I think, mitigate this. Since tab-completion already exists for Code Completion, it feels natural to me (from a user perspective) to extend to Copilot suggestions as well. The other Code Completion trigger options (`$` and `::`) feel less useful in this context, since they're mostly tied to either names of elements in an object or names of functions in a package.

Thanks for the work on the integration (and everything else that makes RStudio so great)!

**Comment by JosiahParry** [comment]
A triggerable code suggestion would be the reason I use copilot in RStudio!  Yes plz <3

**Comment by ronblum** [comment]
Verified in
- RStudio Desktop 2024.04.0-daily+558 on MacOS 14.4
- RStudio Desktop 2024.04.0-daily+555.pro3 on Red Hat 9.3
- Posit Workbench 2024.04.0-daily+555.pro3 on Ubuntu 22.04 via Chrome on MacOS 14.4

Checked both when using global project-based options.

**Comment by JosiahParry** [comment]
@ronblum thank you! If we download a daily install, where might we find this configuration?

**Comment by kevinushey** [comment]
You should see it in Tools -> Global Options... -> Copilot:

![image](https://github.com/rstudio/rstudio/assets/1976582/f329003f-5960-49b4-8f6a-ac09ba60f89a)
